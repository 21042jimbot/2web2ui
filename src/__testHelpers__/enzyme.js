export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
