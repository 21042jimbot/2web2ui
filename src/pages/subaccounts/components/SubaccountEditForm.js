import React, { Component } from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

import { Button, Panel } from '@sparkpost/matchbox';

import { NameField, IpPoolSelect, StatusSelect } from './formFields';

export class SubaccountEditForm extends Component {

  render() {
    const {
      handleSubmit,
      pristine,
      submitting,
      ipPools,
      compliance,
      reset
    } = this.props;

    return (
      <form onSubmit={handleSubmit}>
        <Panel.Section>
          <NameField disabled={submitting || compliance}/>

          { !!ipPools.length && <IpPoolSelect ipPools={ipPools} disabled={submitting || compliance} /> }

          <StatusSelect disabled={submitting || compliance} compliance={compliance} />
        </Panel.Section>
        <Panel.Section>
          <Button submit primary disabled={pristine || submitting || compliance}>
            { submitting ? 'Updating...' : 'Update' }
          </Button>
          { !pristine && <Button style={{ marginLeft: '1em' }} disabled={pristine || submitting} onClick={reset}>Cancel</Button> }
        </Panel.Section>
      </form>
    );
  }
}

const formName = 'SubaccountEditForm';

const mapStateToProps = (state, { subaccount }) => {
  const { compliance } = subaccount;
  // changing the status name if set by compliance because it is uneditable
  const status = compliance ? `${subaccount.status} by SparkPost` : subaccount.status;

  return {
    ipPools: state.ipPools.list,
    compliance,
    initialValues: {
      name: subaccount.name,
      ipPool: subaccount.ip_pool || 'default',
      status
    }
  };
};

const SubaccountEditReduxForm = reduxForm({ form: formName })(SubaccountEditForm);
export default connect(mapStateToProps, {})(SubaccountEditReduxForm);
