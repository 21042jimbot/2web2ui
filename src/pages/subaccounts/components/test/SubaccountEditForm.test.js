import { shallow } from 'enzyme';
import React from 'react';

import { SubaccountEditForm } from '../SubaccountEditForm';

const handleSubmit = jest.fn();

const props = {
  submitting: false,
  pristine: true,
  compliance: false,
  ipPools: ['an array of ip pools'],
  handleSubmit
};

let wrapper;

beforeEach(() => {
  jest.clearAllMocks();
  wrapper = shallow(<SubaccountEditForm {...props} />);
});

test('base case', () => {
  expect(wrapper).toMatchSnapshot();
});

test('no ip pools', () => {
  wrapper.setProps({ ipPools: []});
  expect(wrapper).toMatchSnapshot();
});

test('cancel button', () => {
  wrapper.setProps({ pristine: false });
  expect(wrapper).toMatchSnapshot();
});

it('submits correctly', () => {
  wrapper.find('form').simulate('submit');
  expect(handleSubmit).toHaveBeenCalled();
  wrapper.setProps({ submitting: true });
  expect(wrapper).toMatchSnapshot();
});

it('should disable all fields when compliance is true', () => {
  wrapper.setProps({ compliance: true });
  expect(wrapper).toMatchSnapshot();
});

