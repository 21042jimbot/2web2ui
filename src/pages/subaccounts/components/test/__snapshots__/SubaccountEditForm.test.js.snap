// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base case 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={false}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={false}
          disabled={false}
        />
      </Box>
      <Stack>
        <RestrictToIpPoolCheckbox
          disabled={false}
        />
      </Stack>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={true}
        submit={true}
        variant="primary"
      >
        Update Subaccount
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;

exports[`cancel button 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={false}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={false}
          disabled={false}
        />
      </Box>
      <Stack>
        <RestrictToIpPoolCheckbox
          disabled={false}
        />
      </Stack>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={false}
        submit={true}
        variant="primary"
      >
        Update Subaccount
      </Button>
      <Button
        disabled={false}
        variant="secondary"
      >
        Cancel
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;

exports[`no ip pools 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={false}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={false}
          disabled={false}
        />
      </Box>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={true}
        submit={true}
        variant="primary"
      >
        Update Subaccount
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;

exports[`should disable all fields when compliance is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={true}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={true}
          disabled={true}
        />
      </Box>
      <Stack>
        <RestrictToIpPoolCheckbox
          disabled={true}
        />
      </Stack>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={true}
        submit={true}
        variant="primary"
      >
        Update Subaccount
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;

exports[`submits correctly 1`] = `
<form
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": undefined,
    }
  }
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={true}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={false}
          disabled={true}
        />
      </Box>
      <Stack>
        <RestrictToIpPoolCheckbox
          disabled={true}
        />
      </Stack>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={true}
        submit={true}
        variant="primary"
      >
        Updating...
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;

exports[`with an assigned ip pool 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Panel.Section>
    <Stack>
      <Box
        maxWidth="45rem"
      >
        <NameField
          disabled={false}
        />
      </Box>
      <Box
        maxWidth="45rem"
      >
        <StatusSelect
          compliance={false}
          disabled={false}
        />
      </Box>
      <Stack>
        <RestrictToIpPoolCheckbox
          disabled={false}
        />
        <IpPoolSelect
          disabled={false}
          options={
            Array [
              "an array of ip pools",
            ]
          }
        />
      </Stack>
    </Stack>
  </Panel.Section>
  <Panel.Section>
    <ButtonWrapper>
      <Button
        disabled={true}
        submit={true}
        variant="primary"
      >
        Update Subaccount
      </Button>
    </ButtonWrapper>
  </Panel.Section>
</form>
`;
